src/Common/src/Internal/Runtime/CompilerHelpers/StartupCodeHelpers.cs

```diff
@@ -86,6 +86,22 @@ private static unsafe void InitializeGlobalTablesForModule(IntPtr moduleManager)
                 Debug.Assert(length % IntPtr.Size == 0);
                 InitializeStatics(staticsSection, length);
             }
+
+            // Initialize frozen object segment with GC present
+            IntPtr frozenObjectSection = GetModuleSection(moduleManager, ReadyToRunSectionType.FrozenObjectRegion, out length);
+            if (frozenObjectSection != IntPtr.Zero)
+            {
+                InitializeFrozenObjectSegment(frozenObjectSection, length);
```

Comment:
@nattress There was a `Debug.Assert(length % IntPtr.Size == 0);` on the TFS side, but I don't know why that isn't asserting all the time - I can't find a place we would guarantee the segment size will be divisible by pointer size. I would expect the GC to assert if it needs it, but it doesn't seem to be the case.